clusterman/autoscaler/autoscaler.py

```diff
@@ -174,6 +174,7 @@ def run(self, dry_run: bool = False, timestamp: Optional[arrow.Arrow] = None) ->
             self.target_capacity_gauge.set(new_target_capacity, {"dry_run": dry_run})
             self._emit_requested_resource_metrics(resource_request, dry_run=dry_run)
 
+        self.pool_manager.terminate_expired_orphan_instances(dry_run=dry_run)
```

Comment:
This method can fail due API limit restrictions or when we try to delete a large number of instances. I think we should add exception handling here and gracefully skip this step if we can’t terminate the instance ( We should be able to try terminating the instanct in next run).

we shouldn’t add this method in critical path of scaling since it deals with few exceptional case, so by wrapping in exception we can continue autoscaling the cluster in case of an error in method.

Reply:
Also, It will be safer to enable this behind a feature flag ( using srv_config).

For example it can look something like

if recycle_orphan_enable:
    self.pool_manager.terminate_expired_orphan_instances

we can use the feature toggle to slowly test this in different pools.